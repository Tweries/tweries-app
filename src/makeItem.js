function getId() {
  return Math.round(Math.random() * 36 ** 12).toString(36);
}

function makeItem(hashtags, id, length, position, source = '') {
  return {
    id: id || getId(),
    position,
    source,
    tweet: source
      ? `${position}/${length} ${source} ${hashtags}`
      : `${position}/${length} ${hashtags}`
  };
}

export default makeItem;
