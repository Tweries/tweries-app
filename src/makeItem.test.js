import makeItem from './makeItem';

const HASHTAGS = '#prodmgmt';

describe('makeItem', () => {
  it('Foo', () => {
    expect(makeItem(HASHTAGS, null, 4, 1, 'Foo')).toMatchSnapshot({
      id: expect.any(String)
    });
  });
});
