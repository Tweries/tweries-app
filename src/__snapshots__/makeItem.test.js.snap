// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`makeItem Foo 1`] = `
Object {
  "id": Any<String>,
  "position": 1,
  "source": "Foo",
  "tweet": "1/4 Foo #prodmgmt",
}
`;
